subprojects/core/src/main/java/org/gradle/api/internal/initialization/transform/MergeInstrumentationAnalysisTransform.java

```diff
@@ -108,16 +117,24 @@ public void transform(TransformOutputs outputs) {
             }
         }
 
-        createInstrumentationClasspathMarker(outputs);
         File outputDir = outputs.dir(MERGE_OUTPUT_DIR);
+        createInstrumentationClasspathMarker(outputDir);
         File output = new File(outputDir, DEPENDENCIES_SUPER_TYPES_FILE_NAME);
         serializer.writeTypesMap(output, dependenciesSuperTypes);
-        copyUnchecked(new File(input, METADATA_FILE_NAME), new File(outputDir, METADATA_FILE_NAME));
+        GFileUtils.copyFile(new File(input, METADATA_FILE_NAME), new File(outputDir, METADATA_FILE_NAME));
     }
 
     private InstrumentationTypeRegistry getInstrumentationTypeRegistry() {
         long contextId = getParameters().getContextId().get();
         CacheInstrumentationDataBuildService buildService = getParameters().getBuildService().get();
         return buildService.getInstrumentationTypeRegistry(contextId);
     }
+
+    private static void doOutputOriginalArtifact(File input, TransformOutputs outputs) {
+        if (input.isDirectory()) {
+            outputs.dir(input);
+        } else {
+            outputs.file(input);
+        }
+    }
```

Comment:
🤔 We should at least move this to some utility class.

Reply:
Moved to instrumentation utils